internal/controller/pdbwatcher_controller.go

```diff
@@ -102,14 +122,39 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	deployment := &appsv1.Deployment{}
 	err = r.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: pdbWatcher.Namespace}, deployment)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, err // Error fetching Deployment
 	}
 
+	// Track initial state to detect conflicts
+	initialResourceVersion := deployment.ResourceVersion
+	initialReplicas := *deployment.Spec.Replicas
+
 	// Update status dynamically
-	pdbWatcher.Status.CurrentReplicas = *deployment.Spec.Replicas
-	pdbWatcher.Status.MinReplicas = *deployment.Spec.Replicas
-	pdbWatcher.Status.MaxReplicas = *deployment.Spec.Replicas + deployment.Spec.Strategy.RollingUpdate.MaxSurge.IntVal
-	pdbWatcher.Status.ScaleFactor = 1 // Dynamically Calculated
+	if pdbWatcher.Status.CurrentReplicas == 0 {
+		pdbWatcher.Status.CurrentReplicas = initialReplicas
+	}
+	pdbWatcher.Status.MinReplicas = initialReplicas
+
+	// Handle nil Deployment Strategy and MaxSurge
+	maxSurge := int32(1) // Default max surge value
+	if deployment.Spec.Strategy.RollingUpdate != nil && deployment.Spec.Strategy.RollingUpdate.MaxSurge != nil {
+		if deployment.Spec.Strategy.RollingUpdate.MaxSurge.Type == intstr.Int {
+			maxSurge = deployment.Spec.Strategy.RollingUpdate.MaxSurge.IntVal
+		} else if deployment.Spec.Strategy.RollingUpdate.MaxSurge.Type == intstr.String {
+			percentageStr := strings.TrimSuffix(deployment.Spec.Strategy.RollingUpdate.MaxSurge.StrVal, "%")
+			percentage, err := strconv.Atoi(percentageStr)
+			if err == nil {
+				maxSurge = (initialReplicas * int32(percentage)) / 100
+			}
+		}
+	}
+	pdbWatcher.Status.MaxReplicas = initialReplicas + maxSurge
```

Comment:
your max surge calculation is correct but should this use intiial replicas? You may have already surged? should it instead be MinReplicas + max surge (and see my comment above about not chaning min unless someething else modifies)

Reply:
As above yoy want to use min replicas here or you've lost your initial starting place.

Reply:
if my original replicas was 40 and surge was 10%  I don't want to go over 44 replicas but here you can go infititely big.